tests/everflow/everflow_test_utilities.py

```diff
@@ -729,30 +973,34 @@ def send_and_check_mirror_packets(self,
                 # but DMAC and checksum are trickier. For now, update the TTL and SMAC, and
                 # mask off the DMAC and IP Checksum to verify the packet contents.
                 if self.mirror_type() == "egress":
-                    mirror_packet[packet.IP].ttl -= 1
-                    mirror_packet[packet.Ether].src = setup["router_mac"]
-
-                    inner_packet.set_do_not_care_scapy(packet.Ether, "dst")
+                    if packet.IPv6 in mirror_packet:
+                        mirror_packet[packet.IPv6].hlim -= 1
+                    else:
+                        mirror_packet[packet.IP].ttl -= 1
                     inner_packet.set_do_not_care_scapy(packet.IP, "chksum")
+                    mirror_packet[packet.Ether].src = egress_mirror_src_mac
+                    inner_packet.set_do_not_care_scapy(packet.Ether, "dst")
 
                 logging.info("Expected inner packet: %s", mirror_packet.summary())
                 pytest_assert(inner_packet.pkt_match(mirror_packet), "Mirror payload does not match received packet")
             else:
                 testutils.verify_no_packet_any(ptfadapter, expected_mirror_packet, dest_ports)
 
     @staticmethod
-    def get_expected_mirror_packet(mirror_session, setup, duthost, mirror_packet, check_ttl):
+    def get_expected_mirror_packet(mirror_session, setup, duthost, mirror_packet, check_ttl, router_mac=None):
```

Comment:
Do we need this additional argument router_mac ? As I see the original parameter is also set from the setup as following gre_pkt_src_mac = setup['router_mac']